common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Clients/MsalUserTokenClient.cs

```diff
@@ -0,0 +1,305 @@
+            if (request == null) {
+                return null;
+            }
+            var redirectUri = new UriBuilder(request.Scheme, request.Host.Host,
+                request.Host.Port ?? 0, request.PathBase + "/signin-oidc").ToString();
+            return redirectUri;
+        }
+
+        /// <summary>
+        /// Get scopes
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="scopes"></param>
+        /// <returns></returns>
+        private IEnumerable<string> GetScopes(IOAuthClientConfig config,
+            IEnumerable<string> scopes) {
+            var requestedScopes = new HashSet<string>();
+            if (scopes != null) {
+                foreach (var scope in scopes.Except(ScopesRequestedByMsal)) {
+                    requestedScopes.Add(scope);
+                }
+            }
+            if (config.Audience != null) {
+                requestedScopes.Add(config.Audience + "/.default");
+            }
+            return requestedScopes;
+        }
+
+        /// <summary>
+        /// Get account info for the user
+        /// </summary>
+        /// <param name="application"></param>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public static async Task<IAccount> GetUserAccountAsync(IClientApplicationBase application,
+            ClaimsPrincipal user) {
+            var accountId = user.GetMsalAccountId();
+            IAccount account = null;
+            if (accountId != null) {
+                account = await application.GetAccountAsync(accountId);
+                // Special case for guest users as the Guest oid / tenant id are not surfaced.
+                if (account == null) {
+                    var loginHint = user.GetLoginHint();
+                    if (loginHint == null) {
+                        throw new ArgumentNullException(nameof(loginHint));
+                    }
+                    var accounts = await application.GetAccountsAsync();
+                    account = accounts.FirstOrDefault(a => a.Username == loginHint);
+                }
+            }
+            return account;
+        }
+
+        /// <summary>
+        /// Create public client
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="config"></param>
+        /// <param name="redirectUri"></param>
+        /// <returns></returns>
+        private MsalConfidentialClientDecorator CreateConfidentialClientApplication(
+            ClaimsPrincipal user, IOAuthClientConfig config, string redirectUri = null) {
+            var builder = ConfidentialClientApplicationBuilder.Create(config.ClientId);
+            if (redirectUri != null) {
+                builder = builder.WithRedirectUri(redirectUri);
+            }
+            builder = builder
+                .WithClientSecret(config.ClientSecret)
+                .WithTenantId(config.TenantId)
+              //  .WithHttpClientFactory(...)
```

Comment:
>//  .WithHttpClientFactory(...) [](start = 14, length = 31)

do we still need this commented?

Reply:
Should add a todo:  Added as reminder to pull through http client factory here and use ours instead of default one.